src/libraries/System.Reflection.Emit/tests/PersistableAssemblyBuilder/AssemblySaveTestsWithVariousTypes.cs

```diff
@@ -0,0 +1,455 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.IO;
+using System.Linq;
+using Xunit;
+
+namespace System.Reflection.Emit.Tests
+{
+    [ConditionalClass(typeof(PlatformDetection), nameof(PlatformDetection.IsNotBrowser))]
+    public class AssemblySaveTestsWithVariousTypes
+    {
+        private static readonly AssemblyName s_assemblyName = new AssemblyName("MyDynamicAssembly")
+        {
+            Version = new Version("1.2.3.4"),
+            CultureInfo = Globalization.CultureInfo.CurrentCulture,
+            ContentType = AssemblyContentType.WindowsRuntime,
+            Flags = AssemblyNameFlags.EnableJITcompileTracking | AssemblyNameFlags.Retargetable,
+        };
+
+        [Fact]
+        public void EmptyAssemblyAndModuleTest()
+        {
+            using (TempFile file = TempFile.Create())
+            {
+                AssemblyTools.WriteAssemblyToDisk(s_assemblyName, Type.EmptyTypes, file.Path);
+
+                Assembly assemblyFromDisk = AssemblyTools.TryLoadAssembly(file.Path);
+                Assert.NotNull(assemblyFromDisk);
+
+                AssemblyName aNameFromDisk = assemblyFromDisk.GetName();
+
+                // Test AssemblyName properties
+                Assert.Equal(s_assemblyName.Name, aNameFromDisk.Name);
+                Assert.Equal(s_assemblyName.Version, aNameFromDisk.Version);
+                Assert.Equal(s_assemblyName.CultureInfo, aNameFromDisk.CultureInfo);
+                Assert.Equal(s_assemblyName.CultureName, aNameFromDisk.CultureName);
+                Assert.Equal(s_assemblyName.ContentType, aNameFromDisk.ContentType);
+                // Runtime assemblies adding AssemblyNameFlags.PublicKey in Assembly.GetName() overloads
+                Assert.Equal(s_assemblyName.Flags | AssemblyNameFlags.PublicKey, aNameFromDisk.Flags);
+
+                Module moduleFromDisk = assemblyFromDisk.Modules.First();
+
+                Assert.NotNull(moduleFromDisk);
+                Assert.Equal(s_assemblyName.Name, moduleFromDisk.ScopeName);
+                Assert.Empty(moduleFromDisk.GetTypes());
+            }
+        }
+
+        [Fact]
+        public void OneInterfaceWithMethodsUsingStream()
+        {
+            Type[] types = new Type[] { typeof(IMultipleMethod) };
+            using MemoryStream stream = new MemoryStream();
+
+            // Generate DLL from these and save it to Stream
```

Comment:
nit: this comment and the one below are probably not necessary; the same pattern is repeated elsewhere without comments.